### PR TITLE
feat: Add extensibility support for Windows.Media.Playback.MediaPlayer

### PR DESCRIPTION
**Pull request by jeromelaban**
## PR Type

What kind of change does this PR introduce?
- Feature

## What is the new behavior?

Adds extensibility support for Windows.Media.Playback.MediaPlayer for Skia/Wasm targets.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Review by carldebilly — APPROVED** [review]

File: src/Uno.UWP/Media/Playback/IMediaPlayerExtension.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+using Windows.Foundation;
+using Windows.Storage.Streams;
+using Windows.Media.Playback;
+
+namespace Uno.Media.Playback
+{
+	/// <summary>
+	/// Extension definition for the <see cref="MediaPlayer"/> class
+	/// </summary>
+	public interface IMediaPlayerExtension : IDisposable
```
Comment (dr1rrb) [inline src/Uno.UWP/Media/Playback/IMediaPlayerExtension.cs:15]: XML doc ? 😋

**Review by dr1rrb — APPROVED** [review]

File: src/Uno.UWP/Uno.netcoremobile.csproj [hunk 2]
```diff
@@ -42,6 +42,11 @@
 		<UpToDateCheckInput Include="**\*.cs" Exclude="bin\**\*.cs;obj\**\*.cs;" Visible="False" />
 	</ItemGroup>
 
+	<ItemGroup>
+	  <UpToDateCheckInput Remove="Media\Playback\MediaPlayerRateChangedEventArgs.cs" />
+	  <UpToDateCheckInput Remove="Media\Playback\PlaybackMediaMarkerReachedEventArgs.cs" />
+	</ItemGroup>
```
Comment (Youssef1313) [inline src/Uno.UWP/Uno.netcoremobile.csproj:48]: Are these changes intended?

File: src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlaybackSession.cs [hunk 3]
```diff
@@ -319,7 +319,7 @@ public  bool IsSupportedPlaybackRateRange( double rate1,  double rate2)
 			}
 		}
 		#endif
-		#if __ANDROID__ || __IOS__ || NET461 || __WASM__ || __SKIA__ || __NETSTD_REFERENCE__ || __MACOS__
+		#if __ANDROID__ || __IOS__ || NET461 || false || false || false || __MACOS__
 		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "NET461", "__WASM__", "__SKIA__", "__NETSTD_REFERENCE__", "__MACOS__")]
```
Comment (Youssef1313) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlaybackSession.cs:323]:
```suggestion
		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "NET461")]
```

File: src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs [hunk 4]
```diff
@@ -35,7 +35,7 @@ public  double Volume
 			}
 		}
 		#endif
-		#if __ANDROID__ || __IOS__ || NET461 || __WASM__ || __SKIA__ || __NETSTD_REFERENCE__ || __MACOS__
+		#if __ANDROID__ || __IOS__ || false || false || false || false || __MACOS__
 		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "NET461", "__WASM__", "__SKIA__", "__NETSTD_REFERENCE__", "__MACOS__")]
```
Comment (Youssef1313) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs:39]:
```suggestion
		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "__MACOS__")]
```

File: src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs [hunk 5]
```diff
@@ -49,7 +49,7 @@ public  double PlaybackRate
...
```
Comment (Youssef1313) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs:53]:
```suggestion
		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "__MACOS__")]
```

File: src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs [hunk 6]
```diff
@@ -91,7 +91,7 @@ public  bool AutoPlay
...
```
Comment (Youssef1313) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs:95]:
```suggestion
		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "__MACOS__")]
```

File: src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs [hunk 7]
```diff
@@ -121,7 +121,7 @@ public  bool AutoPlay
...
```
Comment (Youssef1313) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs:125]:
```suggestion
		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "__MACOS__")]
```

File: src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs [hunk 8]
```diff
@@ -141,7 +141,7 @@ public  double BufferingProgress
...
```
Comment (Youssef1313) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs:145]:
```suggestion
		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "__MACOS__")]
```

File: src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs [hunk 9]
```diff
@@ -151,7 +151,7 @@ public  bool CanPause
...
```
Comment (Youssef1313) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs:155]:
```suggestion
		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "__MACOS__")]
```

File: src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs [hunk 10]
```diff
@@ -161,7 +161,7 @@ public  bool CanSeek
...
```
Comment (Youssef1313) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs:165]:
```suggestion
		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "__MACOS__")]
```

File: src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs [hunk 11]
```diff
@@ -175,7 +175,7 @@ public  bool CanSeek
...
```
Comment (Youssef1313) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs:179]:
```suggestion
		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "__MACOS__")]
```

File: src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs [hunk 12]
```diff
@@ -199,7 +199,7 @@ public  bool CanSeek
...
```
Comment (Youssef1313) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs:203]:
```suggestion
		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "__MACOS__")]
```

File: src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs [hunk 13]
```diff
@@ -213,7 +213,7 @@ public  bool CanSeek
...
```
Comment (Youssef1313) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs:217]:
```suggestion
		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "__MACOS__")]
```

File: src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs [hunk 14]
```diff
@@ -241,7 +241,7 @@ public  bool CanSeek
...
```
Comment (Youssef1313) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs:245]:
```suggestion
		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "__MACOS__")]
```

File: src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs [hunk 15]
```diff
@@ -269,7 +269,7 @@ public  bool RealTimePlayback
...
```
Comment (Youssef1313) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs:273]:
```suggestion
		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "__MACOS__")]
```

File: src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs [hunk 16]
```diff
@@ -412,21 +412,21 @@ public MediaPlayer()
...
+		#if false || false || false || NET461 || false || false || false
 		[global::Uno.NotImplemented("NET461", "__WASM__", "__SKIA__", "__NETSTD_REFERENCE__")]
 		public  void Pause()
 		{
 			global::Windows.Foundation.Metadata.ApiInformation.TryRaiseNotImplemented("Windows.Media.Playback.MediaPlayer", "void MediaPlayer.Pause()");
 		}
 		#endif
-		#if __ANDROID__ || __IOS__ || NET461 || __WASM__ || __SKIA__ || __NETSTD_REFERENCE__ || __MACOS__
+		#if __ANDROID__ || __IOS__ || false || false || false || false || __MACOS__
 		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "NET461", "__WASM__", "__SKIA__", "__NETSTD_REFERENCE__", "__MACOS__")]
```
Comment (Youssef1313) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs:430]:
```suggestion
		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "__MACOS__")]
```

File: src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs [hunk 17]
```diff
@@ -435,21 +435,21 @@ public  void SetUriSource( global::System.Uri value)
 		#endif
 		// Forced skipping of method Windows.Media.Playback.MediaPlayer.ProtectionManager.get
 		// Forced skipping of method Windows.Media.Playback.MediaPlayer.ProtectionManager.set
-		#if __ANDROID__ || __IOS__ || NET461 || __WASM__ || __SKIA__ || __NETSTD_REFERENCE__ || __MACOS__
+		#if __ANDROID__ || __IOS__ || false || false || false || false || __MACOS__
 		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "NET461", "__WASM__", "__SKIA__", "__NETSTD_REFERENCE__", "__MACOS__")]
```
Comment (Youssef1313) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs:439]:
```suggestion
		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "__MACOS__")]
```

File: src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs [hunk 18]
```diff
@@ -435,21 +435,21 @@ public  void SetUriSource( global::System.Uri value)
...
 		public  void SetFileSource( global::Windows.Storage.IStorageFile file)
 		{
 			global::Windows.Foundation.Metadata.ApiInformation.TryRaiseNotImplemented("Windows.Media.Playback.MediaPlayer", "void MediaPlayer.SetFileSource(IStorageFile file)");
 		}
 		#endif
-		#if __ANDROID__ || __IOS__ || NET461 || __WASM__ || __SKIA__ || __NETSTD_REFERENCE__ || __MACOS__
+		#if __ANDROID__ || __IOS__ || false || false || false || false || __MACOS__
 		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "NET461", "__WASM__", "__SKIA__", "__NETSTD_REFERENCE__", "__MACOS__")]
```
Comment (Youssef1313) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs:446]:
```suggestion
		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "__MACOS__")]
```

File: src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs [hunk 19]
```diff
@@ -503,14 +503,14 @@ public  void Dispose()
 		// Forced skipping of method Windows.Media.Playback.MediaPlayer.TimelineControllerPositionOffset.get
 		// Forced skipping of method Windows.Media.Playback.MediaPlayer.TimelineControllerPositionOffset.set
 		// Forced skipping of method Windows.Media.Playback.MediaPlayer.PlaybackSession.get
-		#if __ANDROID__ || __IOS__ || NET461 || __WASM__ || __SKIA__ || __NETSTD_REFERENCE__ || __MACOS__
+		#if __ANDROID__ || __IOS__ || false || false || false || false || __MACOS__
 		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "NET461", "__WASM__", "__SKIA__", "__NETSTD_REFERENCE__", "__MACOS__")]
```
Comment (Youssef1313) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs:507]:
```suggestion
		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "__MACOS__")]
```

File: src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs [hunk 20]
```diff
@@ -503,14 +503,14 @@ public  void Dispose()
...
 		public  void StepForwardOneFrame()
 		{
 			global::Windows.Foundation.Metadata.ApiInformation.TryRaiseNotImplemented("Windows.Media.Playback.MediaPlayer", "void MediaPlayer.StepForwardOneFrame()");
 		}
 		#endif
-		#if __ANDROID__ || __IOS__ || NET461 || __WASM__ || __SKIA__ || __NETSTD_REFERENCE__ || __MACOS__
+		#if __ANDROID__ || __IOS__ || false || false || false || false || __MACOS__
 		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "NET461", "__WASM__", "__SKIA__", "__NETSTD_REFERENCE__", "__MACOS__")]
```
Comment (Youssef1313) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs:514]:
```suggestion
		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "__MACOS__")]
```

File: src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs [hunk 21]
```diff
@@ -524,7 +524,7 @@ public  void StepBackwardOneFrame()
 			throw new global::System.NotImplementedException("The member CastingSource MediaPlayer.GetAsCastingSource() is not implemented. For more information, visit https://aka.platform.uno/notimplemented?m=CastingSource%20MediaPlayer.GetAsCastingSource%28%29");
 		}
 		#endif
-		#if __ANDROID__ || __IOS__ || NET461 || __WASM__ || __SKIA__ || __NETSTD_REFERENCE__ || __MACOS__
+		#if __ANDROID__ || __IOS__ || false || false || false || false || __MACOS__
 		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "NET461", "__WASM__", "__SKIA__", "__NETSTD_REFERENCE__", "__MACOS__")]
```
Comment (Youssef1313) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs:528]:
```suggestion
		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "__MACOS__")]
```

File: src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs [hunk 22]
```diff
@@ -581,7 +581,7 @@ public  bool RenderSubtitlesToSurface( global::Windows.Graphics.DirectX.Direct3D
 		}
 		#endif
 		// Forced skipping of method Windows.Media.Playback.MediaPlayer.AudioStateMonitor.get
-		#if __ANDROID__ || __IOS__ || NET461 || __WASM__ || __SKIA__ || __NETSTD_REFERENCE__ || __MACOS__
+		#if __ANDROID__ || __IOS__ || false || false || false || false || __MACOS__
 		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "NET461", "__WASM__", "__SKIA__", "__NETSTD_REFERENCE__", "__MACOS__")]
```
Comment (Youssef1313) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs:585]:
```suggestion
		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "__MACOS__")]
```

File: src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs [hunk 23]
```diff
@@ -597,7 +597,7 @@ public  bool RenderSubtitlesToSurface( global::Windows.Graphics.DirectX.Direct3D
 			}
 		}
 		#endif
-		#if __ANDROID__ || __IOS__ || NET461 || __WASM__ || __SKIA__ || __NETSTD_REFERENCE__ || __MACOS__
+		#if __ANDROID__ || __IOS__ || false || false || false || false || __MACOS__
 		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "NET461", "__WASM__", "__SKIA__", "__NETSTD_REFERENCE__", "__MACOS__")]
```
Comment (Youssef1313) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs:601]:
```suggestion
		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "__MACOS__")]
```

File: src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs [hunk 24]
```diff
@@ -677,7 +677,7 @@ public  bool RenderSubtitlesToSurface( global::Windows.Graphics.DirectX.Direct3D
...
```
Comment (Youssef1313) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs:681]:
```suggestion
		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "__MACOS__")]
```

File: src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs [hunk 25]
```diff
@@ -693,7 +693,7 @@ public  bool RenderSubtitlesToSurface( global::Windows.Graphics.DirectX.Direct3D
...
```
Comment (Youssef1313) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs:697]:
```suggestion
		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "__MACOS__")]
```

File: src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs [hunk 26]
```diff
@@ -773,7 +773,7 @@ public  bool RenderSubtitlesToSurface( global::Windows.Graphics.DirectX.Direct3D
...
```
Comment (Youssef1313) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs:777]:
```suggestion
		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "__MACOS__")]
```

File: src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs [hunk 27]
```diff
@@ -789,7 +789,7 @@ public  bool RenderSubtitlesToSurface( global::Windows.Graphics.DirectX.Direct3D
...
```
Comment (Youssef1313) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.Media.Playback/MediaPlayer.cs:793]:
```suggestion
		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "__MACOS__")]
```
